core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -307,24 +309,33 @@ trait WorkspaceComponent {
 
       loadWorkspaces(workspaces)
     }
-    def listWorkspaceGoogleProjectsToUpdateWithNewBillingAccount(): ReadAction[Seq[(GoogleProjectId, Option[RawlsBillingAccountName])]] = {
+
+    def listWorkspaceGoogleProjectsToUpdateWithNewBillingAccount(): ReadAction[Set[(GoogleProjectId, Option[RawlsBillingAccountName])]] = {
       val query = for {
         billingProject <- rawlsBillingProjectQuery if !billingProject.invalidBillingAccount
         workspace <- workspaceQuery if workspace.namespace === billingProject.projectName &&
-          workspace.workspaceVersion === WorkspaceVersions.V2.value &&
+          workspace.billingAccountErrorMessage.isEmpty &&
           !(workspace.currentBillingAccountOnGoogleProject === billingProject.billingAccount ||
             (workspace.currentBillingAccountOnGoogleProject.isEmpty && billingProject.billingAccount.isEmpty))
       } yield (workspace.googleProjectId, billingProject.billingAccount)
       query.result.map(results => results.map {
         case (googleProjectId, newBillingAccount) =>
           (GoogleProjectId(googleProjectId), newBillingAccount.map(RawlsBillingAccountName))
-      })
+      }.toSet)
     }
 
     def updateWorkspaceBillingAccount(googleProjectId: GoogleProjectId, newBillingAccount: Option[RawlsBillingAccountName]): WriteAction[Int] = {
       findByGoogleProjectIdQuery(googleProjectId).map(_.currentBillingAccountOnGoogleProject).update(newBillingAccount.map(_.value))
     }
 
+    def updateWorkspaceBillingAccountErrorMessage(googleProjectId: GoogleProjectId, errorMessage: String): WriteAction[Int] = {
+      findByGoogleProjectIdQuery(googleProjectId).map(_.billingAccountErrorMessage).update(Option(errorMessage))
```

Comment:
> This finds all the Workspaces in a **Google** project and then then sets them all to have the same error message. That will be problematic for v2 Workspaces.

This should work for both v1 and v2 workspaces since it finds workspaces by their Google project id, not Terra billing project